src/colorize.cr

```diff
@@ -226,120 +220,147 @@ module Colorize
-  private MODE_BRIGHT_FLAG    =  1
-  private MODE_DIM_FLAG       =  2
-  private MODE_UNDERLINE_FLAG =  4
-  private MODE_BLINK_FLAG     =  8
-  private MODE_REVERSE_FLAG   = 16
-  private MODE_HIDDEN_FLAG    = 32
-
-  private COLORS = %w(default black red green yellow blue magenta cyan light_gray dark_gray light_red light_green light_yellow light_blue light_magenta light_cyan white)
-  private MODES  = %w(bold bright dim underline blink reverse hidden)
-
-  @fore : Color
-  @back : Color
-
   def initialize(@object : T)
     @fore = ColorANSI::Default
     @back = ColorANSI::Default
-    @mode = 0
+    @mode = Mode::None
     @enabled = Colorize.enabled?
   end
 
-  {% for name in COLORS %}
-    def {{name.id}}
-      @fore = ColorANSI::{{name.camelcase.id}}
+  {% for color in ColorANSI.constants.reject { |constant| constant == "All" || constant == "None" } %}
+    def {{color.underscore.id}}
+      @fore = ColorANSI::{{color.id}}
       self
     end
 
-    def on_{{name.id}}
-      @back = ColorANSI::{{name.camelcase.id}}
+    def on_{{color.underscore.id}}
+      @back = ColorANSI::{{color.id}}
       self
     end
   {% end %}
 
-  {% for name in MODES %}
-    def {{name.id}}
-      @mode |= MODE_{{name.upcase.id}}_FLAG
+  def on(color : Color)
+    back color
+  end
+
+  {% for mode in Mode.constants.reject { |constant| constant == "All" || constant == "None" } %}
+    def {{mode.underscore.id}}
+      @mode |= Mode::{{mode.id}}
       self
     end
   {% end %}
 
-  def fore(color : Symbol)
-    {% for name in COLORS %}
-      if color == :{{name.id}}
-        @fore = ColorANSI::{{name.camelcase.id}}
-        return self
-      end
-    {% end %}
-
-    raise ArgumentError.new "Unknown color: #{color}"
-  end
-
-  def fore(@fore : Color)
+  @[Deprecated("Please use `bold` instead.")]
+  def bright
+    @mode |= Mode::Bold
     self
   end
 
-  def back(color : Symbol)
-    {% for name in COLORS %}
-      if color == :{{name.id}}
-        @back = ColorANSI::{{name.camelcase.id}}
-        return self
-      end
-    {% end %}
-
-    raise ArgumentError.new "Unknown color: #{color}"
+  # Sets the foreground color of the object to *color*.
+  def fore(color : Color)
+    @fore = color
+    self
   end
 
-  def back(@back : Color)
+  # Sets the background color of the object to *color*.
+  def back(color : Color)
+    @back = color
     self
   end
 
-  def mode(mode : Symbol)
-    {% for name in MODES %}
-      if mode == :{{name.id}}
-        @mode |= MODE_{{name.upcase.id}}_FLAG
-        return self
-      end
-    {% end %}
+  @@warning_printed = false
 
-    raise ArgumentError.new "Unknown mode: #{mode}"
-  end
+  # Adds *mode* to the text's decorations.
+  def mode(mode : Mode)
+    # TODO: Remove this in the future
+    if mode == Mode::Bright
+      puts "Warning: The text decoration `bright` is deprecated. Please use `bold` instead.".colorize(:light_yellow) unless @@warning_printed
+      @@warning_printed = true
+    end
```

Comment:
Do deprecation warnings work on enum members?  Don't think the annotation existed when this PR was made but that would prob be more ideal.

Reply:
Yes, it would be nice if that worked but they don't work.